Parser/Types/TypeAnalyze.cs

```diff
@@ -136,19 +156,51 @@ public override NType Visit (NUnary u)
    public override NType Visit (NIdentifier d) {
       if (mSymbols.Find (d.Name.Text) is NVarDecl v) 
          return d.Type = v.Type;
+      if (mSymbols.Find (d.Name.Text) is NConstDecl c) {
+         ((NLiteral)c.Expr).Accept (this);
+         return d.Type = c.Expr.Type;
+      }
       throw new ParseException (d.Name, "Unknown variable");
    }
 
+   // Check for types.
+   // If types match fine, otherwise do typecase whereever possible.
+   // Helper function, source type and target type is it possible to add a cast.
+   // Throw exception if not parameters match.
+   // Return type assignment already handled.
    public override NType Visit (NFnCall f) {
-      throw new NotImplementedException ();
+      if (mSymbols.Find (f.Name.Text) is NFnDecl fd) {
+         return f.Type = AddFnParamTypes (fd, f.Params, f.Name);
+      }
+      throw new ParseException (f.Name, $"Function {f.Name.Text} does not exist in the current context");
+   }
+
+   // TODO: Need to support method overload.
+   // e.g. Source:Max(int,int) to Target:Max(double,double)
+   NType AddFnParamTypes (NFnDecl fnDecl, NExpr[] exprs, Token token) {
+      if (fnDecl.Params.Length != exprs.Length) {
+         throw new ParseException (token, $"No overload for function '{token.Text}' takes {exprs.Length} aruguments");
+      }
+      for (int i = 0; i < exprs.Length; i++) {
+         var srcType = exprs[i].Accept (this);
+         var target = fnDecl.Params[i];
+         // Type cast should be possible in function call, for int to real.
+         var iConversionPossible = srcType == Int && target.Type == Real;
+         if (!iConversionPossible && srcType != target.Type) {
+            throw new ParseException (token, $"Argument{i + 1}: Cannot convert from '{srcType}' to '{target.Type}'");
+         }

```

Comment:
Other typecasts are also possible:
char => int, char => real, etc...
All the typecast checks should be in one place, since you need to use the same checks also during an assignment to a variable.